core/src/main/java/org/apache/spark/shuffle/api/ShuffleReadSupport.java

```diff
@@ -17,8 +17,15 @@
 
 package org.apache.spark.shuffle.api;
 
+import org.apache.spark.storage.BlockId;
+import scala.Tuple2;
+import scala.collection.Seq;
+
+import java.io.InputStream;
+import java.util.Iterator;
+
 public interface ShuffleReadSupport {
 
-  ShufflePartitionReader newPartitionReader(String appId, int shuffleId, int mapId);
+  Iterator<InputStream> fetchBlocks(String appId, int shuffleId, Iterator<Tuple2<Integer, Seq<BlockId>>> blocks);
```

Comment:
Couple of comments about this API:

- Instead of using `Tuple2` we should make a wrapper class.
- Instead of using `Seq` we should use `java.util.List`.
` This makes `BlockId` part of the public API. Are there implications from that?